lib/util.js

```diff
@@ -403,8 +403,10 @@ function formatValue(ctx, value, recurseTimes) {
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (typeof value === 'function') {
-      return ctx.stylize(`[Function${value.name ? `: ${value.name}` : ''}]`,
-                         'special');
+      const asyncPrefix = binding.isAsyncFunction(value) ? 'async ' : '';
+      return ctx.stylize(
+        `[${asyncPrefix}Function${value.name ? `: ${value.name}` : ''}]`,
```

Comment:
> this will be semver-major

You mean, because it changes the output for `GeneratorFunction`? Yeah, that’s probably for the best. And yes, I think extending it to `GeneratorFunction`s would be a good idea, too.